src/adapter/src/catalog/open.rs

```diff
@@ -1268,6 +1266,16 @@ impl Catalog {
                         }
                     }
                 }
+                CatalogItem::Table(_)
+                | CatalogItem::Source(_)
+                | CatalogItem::MaterializedView(_) => {
+                    // Storage objects don't have any external objects to drop.
+                }
+                CatalogItem::Sink(_) => {
+                    // Sinks don't have any external objects to drop--however,
+                    // this would change if we add a collections for sinks
+                    // #17672.
+                }
```

Comment:
What's going on here? I'm surprised to see this change.

Reply:
If you look right above this, I lump everything together into whether or not the item corresponds to a storage collection--that method unifies everything we need. We also didn't need to understand if a sink has dropped (idk what a builtin sink would be) because we don't currently durably record  any metadata about the sink.